application/org.openjdk.jmc.console.agent/src/main/java/org/openjdk/jmc/console/agent/manager/wizards/CapturedValueEditingPage.java

```diff
@@ -0,0 +1,247 @@
+
+		converterText = createTextInput(container, cols, Messages.CapturedValueEditingPage_LABEL_CONVERTER,
+				Messages.CapturedValueEditingPage_MESSAGE_CONVERTER_DESCRIPTION);
+
+		return container;
+	}
+
+	private void bindListeners() {
+		nameText.addModifyListener(
+				handleExceptionIfAny((ModifyListener) e -> capturedValue.setName(nameText.getText())));
+		if (indexSpinner != null) {
+			indexSpinner.addModifyListener(handleExceptionIfAny(
+					(ModifyListener) e -> ((IMethodParameter) capturedValue).setIndex(indexSpinner.getSelection())));
+		}
+		if (isReturnValueButton != null) {
+			isReturnValueButton.addListener(SWT.Selection, e -> {
+				indexSpinner.setEnabled(!isReturnValueButton.getSelection());
+				if (isReturnValueButton.getSelection()) {
+					convertParameterToReturnValue();
+				} else {
+					convertReturnValueToParameter();
+				}
+			});
+		}
+		if (expressionText != null) {
+			expressionText.addModifyListener(handleExceptionIfAny(
+					(ModifyListener) e -> ((IField) capturedValue).setExpression(expressionText.getText())));
+		}
+		descriptionText.addModifyListener(
+				handleExceptionIfAny((ModifyListener) e -> capturedValue.setDescription(descriptionText.getText())));
+		contentTypeCombo.addModifyListener(handleExceptionIfAny(
+				(ModifyListener) e -> capturedValue.setContentType(contentTypeCombo.getSelectionIndex() == -1 ? null
+						: ContentType.valueOf(contentTypeCombo.getText().toUpperCase(Locale.ENGLISH)))));
+		contentTypeClearButton.addListener(SWT.Selection,
+				handleExceptionIfAny((Listener) e -> contentTypeCombo.deselectAll()));
+		relationalKeyText.addModifyListener(
+				handleExceptionIfAny((ModifyListener) e -> capturedValue.setRelationKey(relationalKeyText.getText())));
+		converterText.addModifyListener(
+				handleExceptionIfAny((ModifyListener) e -> capturedValue.setConverter(converterText.getText())));
+	}
+
+	private void populateUi() {
+		setText(nameText, capturedValue.getName());
+		if (indexSpinner != null) {
+			indexSpinner.setSelection(
+					capturedValue instanceof IMethodParameter ? ((IMethodParameter) capturedValue).getIndex() : 0);
+			indexSpinner.setEnabled(capturedValue instanceof IMethodParameter);
+		}
+		if (isReturnValueButton != null) {
+			isReturnValueButton.setSelection(capturedValue instanceof IMethodReturnValue);
+		}
+		if (expressionText != null) {
+			setText(expressionText, ((IField) capturedValue).getExpression());
+		}
+		setText(descriptionText, capturedValue.getDescription());
+		setText(contentTypeCombo,
+				capturedValue.getContentType() == null ? null : capturedValue.getContentType().toString());
+		setText(relationalKeyText, capturedValue.getRelationKey());
+		setText(converterText, capturedValue.getConverter());
+	}
+
+	public ICapturedValue getResult() {
+		return capturedValue;
+	}
+
+	private void convertParameterToReturnValue() {
+		IMethodParameter parameter = (IMethodParameter) capturedValue;
+
+		// TODO: do this is in model instead
```

Comment:
I don't think these todos are too good for non-authors to really know what should needs to be done instead. I'd rephrase them or remove them

Reply:
These TODOs are no longer needed and have been removed now